### PR TITLE
Add github action for publishing docker image 

### PR DESCRIPTION
**Pull request by RunningLeon**
1. add building and pushing docker image github action yaml config
2. modify Dockerfile so as to build docker image correctly.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -5,12 +5,14 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     && rm -rf /var/lib/apt/lists/*
 
 RUN python3 -m pip install torch==1.13.1+cu117 torchvision==0.14.1+cu117 --extra-index-url https://download.pytorch.org/whl/cu117
-RUN python3 -m pip install sentencepiece cmake transformers<4.30.0 protobuf==3.20.3 pybind11 mmengine
+RUN python3 -m pip install sentencepiece cmake "transformers<4.30.0" protobuf==3.20.3 pybind11 mmengine
```
Comment (AllentDan) [inline docker/Dockerfile:8]: transformers<4.30.0 is no longer required. 
all the packages are listed in `requirements.txt` and can be installed through in the following `pip install .`

Comment (RunningLeon) [reply docker/Dockerfile:8]: no protobuf and cmake in https://github.com/InternLM/lmdeploy/blob/main/requirements.txt

Comment (AllentDan) [reply docker/Dockerfile]: `protobuf` can be installed through `tritonclient[all]`

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Get lmdeploy commit id
+        run: |
+          export commit_id=$(git log --format="%H" -n 1)
+          echo "commit id = $commit_id"
+          echo "COMMIT_ID=$commit_id" >> $GITHUB_ENV
+      - name: Get lmdeploy version
+        if: startsWith(github.ref, 'refs/tags/') == false
```
Comment (lvhan028) [inline .github/workflows/docker.yml:32]: How to invoke this action?

Comment (RunningLeon) [reply .github/workflows/docker.yml:32]: These two steps are used to determine the tag of built docker image, which would be either `openmmlab/lmdeploy:latest` or `openmmlab/lmdeploy:v0.x.x`

1.  When a PR is merged into main branch and  this step`startsWith(github.ref, 'refs/tags/') == false` is met
2. When a tag is pushed and this step`startsWith(github.ref, 'refs/tags/') == true` is met

**Review by lvhan028 — APPROVED** [review]

**Review by AllentDan — APPROVED** [review]

**Pull request merged**